src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/AdyenPaymentRequestSender.java

```diff
@@ -29,49 +35,156 @@
+                return paymentPort.cancel(modificationRequest);
+            }
+        });
     }
 
-    public ModificationResult cancel(final String countryIsoCode, final ModificationRequest modificationRequest) throws ServiceException {
-        final PaymentPortType paymentPort = adyenPaymentPortRegistry.getPaymentPort(countryIsoCode);
-        return paymentPort.cancel(modificationRequest);
+    public AdyenCallResult<ModificationResult> cancelOrRefund(final String countryIsoCode, final ModificationRequest modificationRequest) {
+        return callAdyen(countryIsoCode, new AdyenCall<PaymentPortType, ModificationResult>() {
+            @Override
+            public ModificationResult apply(final PaymentPortType paymentPort) throws ServiceException {
+                return paymentPort.cancelOrRefund(modificationRequest);
+            }
+        });
     }
 
-    public ModificationResult cancelOrRefund(final String countryIsoCode, final ModificationRequest modificationRequest) throws ServiceException {
-        final PaymentPortType paymentPort = adyenPaymentPortRegistry.getPaymentPort(countryIsoCode);
-        return paymentPort.cancelOrRefund(modificationRequest);
+    public AdyenCallResult<ModificationResult> capture(final String countryIsoCode, final ModificationRequest modificationRequest) {
+        return callAdyen(countryIsoCode, new AdyenCall<PaymentPortType, ModificationResult>() {
+            @Override
+            public ModificationResult apply(final PaymentPortType paymentPort) throws ServiceException {
+                return paymentPort.capture(modificationRequest);
+            }
+        });
     }
 
-    public ModificationResult capture(final String countryIsoCode, final ModificationRequest modificationRequest) throws ServiceException {
-        final PaymentPortType paymentPort = adyenPaymentPortRegistry.getPaymentPort(countryIsoCode);
-        return paymentPort.capture(modificationRequest);
+    private <T> AdyenCallResult<T> callAdyen(final String countryIsoCode, final AdyenCall<PaymentPortType, T> adyenCall) {
+        try {
+            final PaymentPortType paymentPort = adyenPaymentPortRegistry.getPaymentPort(countryIsoCode);
+            final T result = adyenCall.apply(paymentPort);
+            return new SuccessfulAdyenCall<T>(result);
+        } catch (final Exception e) {
+            logger.info("exception during adyen request", e);
+            return mapExceptionToCallResult(e);
+        }
+    }
+
+    /**
+     * Educated guess approach to transform CXF exceptions into error status codes.
+     * In the future if we encounter further different cases it makes sense to change this if/else structure to a map with lookup.
+     */
+    private <T> AdyenCallResult<T> mapExceptionToCallResult(final Exception e) {
+        //noinspection ThrowableResultOfMethodCallIgnored
+        final Throwable rootCause = Throwables.getRootCause(e);
```

Comment:
Good point! Checked my examples and there is more or less always the root cause message >= outer exception message. I'll change it to rootCause.getMessage()